jcl/source/common/JclSysInfo.pas

```diff
@@ -4293,6 +4294,28 @@ function GetWindowsReleaseName: String;
     Result := '';
 end;
 
+function GetWindowsReleaseCode: String;
+begin
+  // Looks much like the 'GetWindowsReleaseCodeName', except for the Windows 10 versions
+  // prior to Release Id 1903 - those have a different 'code' vs the 'code name'.
+  if IsWin10 then
+  begin
+    case GetWindowsReleaseId of
+      1903:
+        Result := '19H1';
+      1909:
+        Result := '19H2';
+      2004:
+        Result := '20H1';
+      2009:
+        Result := GetWindowsDisplayVersion;
```

Comment:
So this gives 20H2 for this version?

Reply:
I think you mean this regarding '2009' - starting that 'version' all newer versions still have that same 'ReleaseId', but the 'Code' has changed to be the same as the 'DisplayVersion'.

So it will give back 21H1, 21H2, and so on for higher versions of Windows 10 and 11.